src/vmm/src/devices/virtio/rng/device.rs

```diff
@@ -112,15 +112,15 @@ impl Entropy {
             return Ok(0);
         }
 
-        let mut rand_bytes = vec![0; iovec.len() as usize];
+        let mut rand_bytes = vec![0; iovec.len()];
         rand::fill(&mut rand_bytes).map_err(|err| {
             METRICS.host_rng_fails.inc();
             err
         })?;
 
         // It is ok to unwrap here. We are writing `iovec.len()` bytes at offset 0.
         iovec.write_all_volatile_at(&rand_bytes, 0).unwrap();
-        Ok(iovec.len())
+        Ok(u32::try_from(iovec.len()).unwrap())
```

Comment:
this makes me sad :(

I don't think its strictly necessary, but it'd be cool to differentiate between the new "multi descriptor chain" iovecbuffermuts and the old "one descriptor chain" ones. We could add a `L: Add` type parameter that is either `u32` or `usize` and specifies the length of the `IoVecBufferMut`. Then only the `impl IoVecBufferMut<usize>` block would have all the appending stuff, and in the rng device and everywhere else can continue using `IoVecBufferMut<u32>` without these ugly `try_froms`.